draft-ietf-ice-rfc5245bis.xml

```diff
@@ -1611,54 +1420,50 @@ of this specification.
 <section anchor="sec-forming" title="Forming the Check Lists">
 
 <t>
-There is one CHECK LIST per in-use media stream resulting from the
-candidate exchange.  To form the CHECK LIST for a media stream, the
-agent forms candidate pairs, computes a candidate pair priority,
-orders the pairs by priority, prunes them, removes lower-priority
-candidates and sets their states. These steps are described in this section.
-If a CHECK LIST is updated (e.g, due to detection of peer reflexive
-candidates), the agent will re-perform the steps for the updated
-CHECK LIST.
+There is one check list for each media stream.  To form a check list,
+an agent (initiaing and responding) forms candidate pairs, computes
+pair priorities, orders pairs by priority, prunes pairs, removes
+lower-priority pairs, and sets check list states. If candidates are
+added to a check list (e.g, due to detection of peer reflexive
+candidates), the agent will re-perform these steps for the updated
+check list.
 </t>
 <section title="Check List State">
-<t>Each CHECK LIST has a state, which captures the state of ICE checks
-for the media stream associated with the CHECK LIST. The states are:</t>
+<t>Each check list has one of the following states:</t>
 <t><list style="hanging">
-<t hangText="Running:"> In this state, ICE checks are still in
-progress for this media stream. Check lists are initially set
+
+<t hangText="Running:">The check list is neither Completed yet nor
+Failed yet.  Check lists are initially set
 to the Running state.
 </t>
-<t hangText="Completed:"> In this state, ICE checks have produced
-selected pairs for each component of the media stream.
+
+<t hangText="Completed:">The check list has a selected candidate pair
+for each component of the media stream.
 </t>
-<t hangText="Failed:"> In this state, the ICE checks have not
-completed successfully for this media stream.
+
+<t hangText="Failed:">The check list does not have a valid candidate
+pair for each component of the media stream and all of the candidate
+pairs in the check list are in either the Failed or Succeeded state.
+In other words, at least one component of the check list has candidate
+pairs that are all in the Failed state, which means the component has 
+failed, which means the check list has failed.
 </t>
+
 </list></t>
-<t>A CHECK LIST with at least one pair that is
-Waiting is called an active CHECK LIST, and a CHECK LIST with all
-pairs Frozen is called a frozen CHECK LIST.
+<t>Additionally, a check list with at least one pair in the waiting
+state is called "active", while a check list with all pairs in the
+frozen state is called "frozen".
```

Comment:
The trickle spec does talk about active and frozen check lists. For example:


   A Trickle ICE agent initially considers all check lists to be frozen.
   It then inspects the first check list and attempts to unfreeze all
   candidate pairs it has received so far that belong to the first
   component on the first media stream (i.e., the first media stream
   that was reported to the ICE implementation from the using
   application).  If that first component of the first media stream does
   not contain candidates for one or more of the currently known pair
   foundations, and if candidate pairs already exist for that foundation
   in one of the following components or media streams, then the agent
   unfreezes the first of those candidate pairs.

And:


   The ICE specification [rfc5245bis], Section 5.1.4, requires that an
   agent will terminate the timer for a triggered check in relation to
   an active check list once the agent has exhausted all frozen pairs in
   the check list.  This will not work with Trickle ICE, because more
   pairs will be added to the check list incrementally.

And:


   When a check list is set to Failed as described above, regular ICE
   requires the agent to update all other check lists, placing one pair
   from each check list into the Waiting state - effectively unfreezing
   all remaining check lists.  However, under Trickle ICE other check
   lists might still be empty at this point (because candidates have not
   yet been received), and following only the rules from regular ICE
   would prevent the agent from unfreezing those check lists (because
   the state of a check list depends on the state of the candidate pairs
   in that check list, but there are none yet).  Therefore a Trickle ICE
   agent needs to monitor whether a check list is active or frozen
   independently of the state of the candidate pairs in the check list
   (since there might not be any pairs yet).  With regard to empty check
   lists, by default a Trickle ICE agent MAY consider an empty check
   list to be either active or frozen.  When a Trickle ICE agent
   considers an empty check list to be frozen, during the candidate
   checking process it SHOULD change the check list to active if
   checking of another check list is completely finished (i.e., if every
   pair in the other check list is either Successful or Failed), if
   another check list has a valid candidate pair for all components, or
   if it adds a candidate pair to the check list (because, in accordance
   with Section 8.1.1, when inserting a new candidate pair into an empty
   check list, the agent sets the pair to a state of Waiting).

I'll take this to the list.